recipes-core/varstored/varstored/0001-add-oxtdb-varstore.patch

```diff
@@ -0,0 +1,891 @@
++    if (send_credit > MAX_CREDIT)
++        send_credit = MAX_CREDIT;
++
++    if (send_credit > 0) {
++        send_credit--;
++    } else {
++        /* If no credit, wait the correct amount of time to get a credit. */
++        struct timespec ts = {0, NS_PER_CREDIT};
++
++        nanosleep(&ts, NULL);
++        last_time = time(NULL);
++    }
++}
++
++static int oxtdb_dbus_init(void) {
++    GError * err = NULL;
++    DBusGConnection * gdbus_conn;
++
++    if (xcdbus_conn) {
++        return 0;
++    }
++
++    event_init();
++    g_type_init();
++
++    gdbus_conn = dbus_g_bus_get(DBUS_BUS_SYSTEM, &err);
++
++    if (!gdbus_conn) {
++        g_error_free(err);
++        return -1;
++    }
++
++    xcdbus_conn = xcdbus_init_event("com.citrix.xenclient.db", gdbus_conn);
++    if (xcdbus_conn) {
++        return -1;
++    }
++
++    return 0;
++}
++
++// Write a value to the specified DB path.
++static void oxtdb_write(char * value, const char * path_format, ...) {
++
++    va_list args;
++    char * path = malloc(MAX_OXTDB_PATH_SIZE);
++
++    oxtdb_dbus_init();
++
++    va_start(args, path_format);
++    vsnprintf(path, MAX_OXTDB_PATH_SIZE, path_format, args);
++    va_end(args);
++
++    if (xcdbus_conn) {
++        com_citrix_xenclient_db_write_(xcdbus_conn, OXTDB_DBUS_SERVICE, OXTDB_DBUS_PATH, path, value);
++    }
++
++    free(path);
++}
++
++// Determine if db node exists.
++static bool oxtdb_exists(const char * path_format, ...) {
++
++    int found = false;
++
++    va_list args;
++    char * path = malloc(MAX_OXTDB_PATH_SIZE);
++
++    oxtdb_dbus_init();
++
++    va_start(args, path_format);
++    vsnprintf(path, MAX_OXTDB_PATH_SIZE, path_format, args);
++    va_end(args);
++
++    if (xcdbus_conn) {
++        com_citrix_xenclient_db_exists_(xcdbus_conn, OXTDB_DBUS_SERVICE, OXTDB_DBUS_PATH, path, &found);
++    }
++
++    free(path);
++
++    return (bool)found;
++}
++
++// Allocates memory!
++// Gets the value of the specified DB key.
++// The string returned should be freed.
++static char * oxtdb_read(const char * path_format, ...) {
++    char * string;
++    va_list args;
++    char * path = malloc(MAX_OXTDB_PATH_SIZE);
++
++    oxtdb_dbus_init();
++
++    va_start(args, path_format);
++    vsnprintf(path, MAX_OXTDB_PATH_SIZE, path_format, args);
++    va_end(args);
++
++    if (xcdbus_conn && com_citrix_xenclient_db_read_(xcdbus_conn, OXTDB_DBUS_SERVICE, OXTDB_DBUS_PATH, path, &string)) {
++        free(path);
++        return string;
++    } else {
++        free(path);
++        return NULL;
++    }
++}
++
++/* The VM's uuid. Used for saving to the db. */
++char *oxtdb_arg_uuid;
++
++size_t
++oxtdb_serialize_variables(uint8_t **out, size_t *len, bool only_nv)
++{
++    return xapidb_serialize_variables(out, len, only_nv);
++}
++
++static bool
++write_all(const char *buf)
++{
++    bool check = false;
++    int var_num = 0;
++    check = oxtdb_exists("/vm/%s/varstored-efivar-blob/%d", oxtdb_arg_uuid, var_num);
```

Comment:
var_num is always 0.  So you could just use "/vm/%s/varstored-efivar-blob/0", but there is no need for sub-keys, so "/vm/%s/varstored-efivar-blob" should suffice.